### ISSUE TITLE
Goaccess takes long time to quit

### ISSUE THREAD

**Issue by plantroon** [allinurl/goaccess#1312]
There is a considerable time delay before goaccess closes once I send SIGINT to it. Is there any way to skip whatever it is doing throughout that time (other than killing it)? I need to start a new instance on the same port as soon as possible. Right now I use a slight time delay, around 2 minutes just to be sure.

I tried using --keep-db-files and it still takes long. I need it to reset at midnight to show statistics for the current day or, better, display only the last 24 hours. None of these two things seem to be possible currently with realtime html.

**Comment by allinurl** [comment]
Can you please post the exact command you are running so I can try to reproduce this from my end? Thanks.

**Comment by plantroon** [comment]
I run it like this with only log format specified in goaccess.conf:
`/usr/bin/goaccess -f /storage/log/nginx/example.com/nginx-access-current.log --real-time-html --ws-url=ws://logs.example.com:80/ws2/ --port=8010 --ignore-crawlers -o /storage/goaccess/index.html --db-path=/storage/goaccess --origin=http://logs.example.com --fifo-in=/tmp/wspipe2.in --fifo-out=/tmp/wspipe2.out`

**Comment by plantroon** [comment]
This was apparently due to using the on disk database that is the default in Debian's package.

I switched to goaccess repo to install the newest goaccess and now it doesn't create any database on the disk, which is good for me. What CLI switch actually controls this, if I may ask? Could you make it more apparent in the manual? For some reason Debian's package does not use in-memory hash tables by default.

**Comment by plantroon** [comment]
What do you mean? I just did apt install goaccess.

**Comment by allinurl** [comment]
My guess is that you are using goaccess with the on-disk storage. You can run `goaccess -s` to find out. I'd try installing goaccess from the [**official deb repo** ](https://github.com/allinurl/goaccess#official-goaccess-debian--ubuntu-repository) and see if that solves the issue. Please let me know how it goes. Thanks.

**Comment by plantroon** [comment]
Yes it did solve the issue. As I said in my last message:

This was apparently due to using the on disk database that is the default in Debian's package.

I switched to goaccess repo to install the newest goaccess and now it doesn't create any database on the disk, which is good for me. What CLI switch actually controls this, if I may ask? Could you make it more apparent in the manual? For some reason Debian's package does not use in-memory hash tables by default.

So basically I just asked if you could make it clearer in the manual how to switch the storage backend because I still honestly don't know. I am just glad that it works :D 

Thanks for the awesome app, I love the visual design.

**Comment by atc0005** [comment]
> @plantroon: I switched to goaccess repo to install the newest goaccess

> @allinurl: How did you install goaccess from the repo?

> @plantroon: What do you mean? I just did apt install goaccess.

---

@plantroon I thinkk what @allinurl  was asking is what *specific* steps you took to enable use of the upstream goaccess repo. I think the goal is to be certain that when you ran `apt install goaccess` the upstream package was installed instead of the package from the Ubuntu repos.

FWIW, I too am seeing goaccess take a long time to quit. In my case I've asked it to resolve IPs, so I suspect that something associated with that task is causing the issue.

I ended up switching out which nameservers my box had (switched from Windows DNS servers to a test dnsmasq instance) and that seems to have helped "unstick" the name resolution issue (goaccess is now at least showing some kind of activity/progress), even though "Closing GoAccess" is displayed and processing appears to continue.

`$ goaccess --version`
```ShellSession
GoAccess - 1.2.
For more details visit: http://goaccess.io
Copyright (C) 2009-2016 by Gerardo Orellana
```

If it helps, this is what I've run:

`$ grep '29/Jan' other_vhosts_access.log | goaccess --log-format VCOMMON --agent-list --with-output-resolver -o SYSTEM_NAME_HERE-2019-01-29.html`

Number of entries in the Apache log file:

`$ wc -l < /tmp/other_vhosts_access.log`
```ShellSession
96011
```
